src/common/typedefs/Resource.d.ts

```diff
@@ -123,7 +129,8 @@ export interface IResource {
   noDelete?: true;
   name: { singular: string; plural: TResourceType };
   ListItem: JSX.Element<any>;
-  getList: (params: IListParams) => Promise<IListResponse>;
+  // getList can now be a hashmap
+  getList: any;
```

Comment:
if it's a hashmap I'd type this as `{[k: string]: WhateverValueTypeHere}`. If it's polymorphic, a union type can be used :)